src/xapi/types.ts

```diff
@@ -0,0 +1,43 @@
+import { Duplex } from 'stream';
+
+export interface CloseableStream extends Duplex {
+  close(): void;
+}
+
+export type Handler = () => void;
+export type Path = string | string[];
+export type NormalizedPath = Array<string | number>;
+export type Listener = (...args: any[]) => void;
+
+export interface XapiOptions {
+  feedbackInterceptor?: never;
+}
+
+export interface XapiRequest {
+  id?: string;
+  method: string;
+  jsonrpc: string;
+  params?: any;
+}
+
+export interface XapiResponse {
+  id: string;
+  method: string;
+  params: any;
+  result: XapiResult;
+  error: XapiError;
+}
```

Comment:
Some of these properties are mutually exclusive. I think in `json-rpc` you either get a `result` or an `error`, depending on the response being a success or a failure. I'm not looking at the spec currently, but I think `params` is optional too.